### PR TITLE
HCK-7614: column DEFAULT

### PR DESCRIPTION
**Pull request by chulanovskyi-bs**
<table>
<td>
  <a href="https://hackolade.atlassian.net/browse/HCK-7614" title="HCK-7614" target="_blank">Sub-taskHCK-7614</a>  Implement ALTER statement for DEFAULT prop
  </td></table>
  <br />
 

## Technical details

* added column DEFAULT prop handler (main [implementation](https://github.com/hackolade/PostgreSQL/compare/feat/HCK-7614-alter-default?expand=1#diff-f3bd950ac23fd4fcabed248722f98c65ea67aab4627f1da040da94676b0162d1R1))
* refactored a bit `utils/general` module, removing the _incoming_ dependency (lodash) and requiring it as usual
* bumped and locked `pg` version, because it used `8.12.0` (latest) anyway

### CODE REVIEW

File: forward_engineering/alterScript/alterScriptHelpers/columnHelpers/defaultValueHelper.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+ * @param { ({ ddlProvider: Object, collection: Object }) }
+ * @returns { Array<AlterScriptDto> }
+ * */
+const getUpdatedDefaultColumnValueScriptDtos = ({ ddlProvider, collection }) =>
+	toPairs(collection.properties)
+		.filter(([_name, jsonSchema]) => {
+			const newDefault = jsonSchema.default;
+			const oldName = jsonSchema.compMod.oldField.name;
+			const oldDefault = collection.role.properties[oldName]?.default;
+			return newDefault && (!oldDefault || newDefault !== oldDefault);
```
Comment (serhii-filonenko) [inline forward_engineering/alterScript/alterScriptHelpers/columnHelpers/defaultValueHelper.js:15]: What if the `newDefault` for the numeric type is 0 ?

Comment (taras-dubyk) [reply forward_engineering/alterScript/alterScriptHelpers/columnHelpers/defaultValueHelper.js:15]: I would also check a boolean column `false` default.

File: forward_engineering/ddlProvider/templates.js [hunk 2]
```diff
@@ -134,4 +134,7 @@ module.exports = {
 	setSequenceType: 'ALTER SEQUENCE IF EXISTS ${name} SET ${sequenceType};\n',
 
 	alterSequence: 'ALTER SEQUENCE IF EXISTS ${name}' + '${options}' + ';\n',
+
+	updateColumnDefaultValue: 'ALTER TABLE ${tableName} ALTER COLUMN ${columnName} SET DEFAULT ${defaultValue};',
```
Comment (serhii-filonenko) [inline forward_engineering/ddlProvider/templates.js:138]: https://github.com/hackolade/PostgreSQL/blob/21ba14993188300d00bd5b308d9feb9122c60114/forward_engineering/ddlProvider/templates.js#L32-L42

The rest of the alter table templates use the `IF EXISTS` clause, perhaps it’s worth adding it here too

File: forward_engineering/alterScript/alterScriptHelpers/columnHelpers/defaultValueHelper.js (hunk 1 above)
Comment (chulanovskyi-bs) [reply forward_engineering/alterScript/alterScriptHelpers/columnHelpers/defaultValueHelper.js]: Good point, updated and verified 👍🏻 

image

**Review by taras-dubyk — APPROVED** [review]

**Pull request merged**